src/charm.py

```diff
@@ -862,19 +863,25 @@ def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
         else:
             self.unit.status = WaitingStatus(PRIMARY_NOT_REACHABLE_MESSAGE)
 
-    def _on_config_changed(self, _) -> None:
+    def _on_config_changed(self, event) -> None:
         """Handle configuration changes, like enabling plugins."""
         if not self.is_cluster_initialised:
-            logger.debug("Early exit on_config_changed: cluster not initialised yet")
+            logger.debug("Defer on_config_changed: cluster not initialised yet")
+            event.defer()
```

Comment:
Deferring, because otherwise we'll never validate the config.

Reply:
Curiosity question to simplify the code:
can we improve the ops lib to print DEBUG message if defer() got comment?
It will be only:
```
event.defer("Defer on_config_changed: cluster not initialized yet")
```